tests/inject/karma.conf.js

```diff
@@ -48,19 +62,52 @@ module.exports = (config) => {
                 sourcemap: 'inline',
             },
         },
-        reporters: ['progress', 'coverage'],
-        coverageReporter: {
-            type: 'html',
-            dir: 'tests/inject/coverage/'
-        },
+        reporters: ['spec'],
         port: 9876,
         colors: true,
         logLevel: config.LOG_INFO,
         autoWatch: true,
-        browsers: config.debug ?
-            ['Chrome'] :
-            ['Chrome', 'Firefox', process.platform === 'darwin' ? 'Safari' : null].filter(Boolean),
-        singleRun: config.debug ? false : true,
-        concurrency: config.debug ? Infinity : 1,
-    });
+        browsers: ['Chrome', 'Firefox', process.platform === 'darwin' ? 'Safari' : null].filter(Boolean),
+        singleRun: true,
+        concurrency: 1,
+    };
+
+    if (config.debug) {
+        options.browsers = ['Chrome'];
+        options.singleRun = false;
+        options.concurrency = Infinity;
+        options.logLevel = config.LOG_DEBUG;
+    }
+
+    if (config.ci) {
+        const {CI_BUILD_CONTEXT, CHROME_BIN, FIREFOX_BIN} = process.env;
+        const configureCIBrowsers = require('./karma.conf.ci');
+        options = {...options, ...configureCIBrowsers(CI_BUILD_CONTEXT, {CHROME_BIN, FIREFOX_BIN})};
+        options.autoWatch = false;
+        options.singleRun = true;
+        options.concurrency = 1;
+        options.logLevel = config.LOG_DEBUG;
+    }
+
+    if (config.coverage) {
+        const plugin = rollupPluginIstanbul({
+            exclude: ['tests/**/*.*', 'src/inject/dynamic-theme/stylesheet-proxy.ts'],
+        });
+        options.rollupPreprocessor.plugins.push(plugin);
+        options.reporters.push('coverage');
+        options.coverageReporter = {
+            type: 'html',
+            dir: 'tests/inject/coverage/'
+        };
+    }
```

Comment:
Moved the coverage stuff down here so we can control it using a flag. It shaves off a few milliseconds per build when you don't need coverage calculated.